File: ios-mma-recomp/MMARecomp/Models/Workout.swift
import Foundation

/// Log réel d'une séance (jambes/torse/HIIT). Le date est en "yyyy-MM-dd"
/// pour matcher directement la colonne Postgres `date` sans souci de fuseau
/// horaire côté décodage JSON.
struct Workout: Codable, Identifiable {
    var id: UUID
    var userId: UUID
    var date: String
    var type: WorkoutType
    var exercices: [LoggedExercise]
    var dureeMin: Int?
    var notes: String?
    var createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case date, type, exercices
        case dureeMin = "duree_min"
        case notes
        case createdAt = "created_at"
    }
}

struct NewWorkout: Codable {
    var date: String
    var type: WorkoutType
    var exercices: [LoggedExercise]
    var dureeMin: Int?
    var notes: String?

    enum CodingKeys: String, CodingKey {
        case date, type, exercices
        case dureeMin = "duree_min"
        case notes
    }
}

File: android-mma-recomp/app/src/main/java/com/example/mmarecomp/model/Workout.kt
package com.example.mmarecomp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Log réel d'une séance (jambes/torse/HIIT). `date` reste en "yyyy-MM-dd"
 *  pour matcher directement la colonne Postgres `date`. */
@Serializable
data class Workout(
    val id: String,
    @SerialName("user_id") val userId: String,
    val date: String,
    val type: WorkoutType,
    val exercices: List<LoggedExercise>,
    @SerialName("duree_min") val dureeMin: Int? = null,
    /** RPE global de la séance (1-10). Multiplié par la durée, il donne la
     *  charge interne (session-RPE) dont dérive l'ACWR. */
    val rpe: Int? = null,
    val notes: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
)

@Serializable
data class NewWorkout(
    val date: String,
    val type: WorkoutType,
    val exercices: List<LoggedExercise>,
    @SerialName("duree_min") val dureeMin: Int? = null,
    val rpe: Int? = null,
    val notes: String? = null,
)
